sdk/src/accounts/webSocketDriftClientAccountSubscriberV2.ts

```diff
@@ -0,0 +1,697 @@
+		await Promise.all([
+			// Perp market subscription
+			this.perpMarketAllAccountsSubscriber.subscribe(
+				(
+					_accountId: PublicKey,
+					data: PerpMarketAccount,
+					context: Context,
+					_buffer: Buffer
+				) => {
+					if (
+						this.delistedMarketSetting !== DelistedMarketSetting.Subscribe &&
+						isVariant(data.status, 'delisted')
+					) {
+						return;
+					}
+					this.perpMarketAccountLatestData.set(data.marketIndex, {
+						data,
+						slot: context.slot,
+					});
+					this.eventEmitter.emit('perpMarketAccountUpdate', data);
+					this.eventEmitter.emit('update');
+				}
+			),
+			// Spot market subscription
+			this.spotMarketAllAccountsSubscriber.subscribe(
+				(
+					_accountId: PublicKey,
+					data: SpotMarketAccount,
+					context: Context,
+					_buffer: Buffer
+				) => {
+					if (
+						this.delistedMarketSetting !== DelistedMarketSetting.Subscribe &&
+						isVariant(data.status, 'delisted')
+					) {
+						return;
+					}
+					this.spotMarketAccountLatestData.set(data.marketIndex, {
+						data,
+						slot: context.slot,
+					});
+					this.eventEmitter.emit('spotMarketAccountUpdate', data);
+					this.eventEmitter.emit('update');
+				}
+			),
+			// State account subscription
+			(async () => {
+				const statePublicKey = await getDriftStateAccountPublicKey(
+					this.program.programId
+				);
+				this.stateAccountSubscriber = new WebSocketAccountSubscriberV2(
+					'state',
+					this.program,
+					statePublicKey,
+					undefined,
+					undefined,
+					this.commitment as Commitment
+				);
+				this.stateAccountSubscriber.subscribe((data: StateAccount) => {
+					this.eventEmitter.emit('stateAccountUpdate', data);
+					this.eventEmitter.emit('update');
+				});
+			})(),
+			(async () => {
+				await this.setInitialData();
+				const subscribeToOraclesStartTime = performance.now();
+				await this.subscribeToOracles();
+				const subscribeToOraclesEndTime = performance.now();
+				const duration =
+					subscribeToOraclesEndTime - subscribeToOraclesStartTime;
+				return duration;
+			})(),
+			this.stateAccountSubscriber.fetch(),
+		]);
+
+		const initialPerpMarketDataFromLatestData = new Map(
+			Array.from(this.perpMarketAccountLatestData.values()).map((data) => [
+				data.data.marketIndex,
+				data.data,
+			])
+		);
+		const initialSpotMarketDataFromLatestData = new Map(
+			Array.from(this.spotMarketAccountLatestData.values()).map((data) => [
+				data.data.marketIndex,
+				data.data,
+			])
+		);
+		this.initialPerpMarketAccountData = initialPerpMarketDataFromLatestData;
+		this.initialSpotMarketAccountData = initialSpotMarketDataFromLatestData;
+
+		this.eventEmitter.emit('update');
+
+		await this.handleDelistedMarketOracles();
+
+		await Promise.all([this.setPerpOracleMap(), this.setSpotOracleMap()]);
+
+		this.isSubscribing = false;
+		this.isSubscribed = true;
+		this.subscriptionPromiseResolver(true);
+
+		// delete initial data
+		this.removeInitialData();
```

Comment:
should this be here?

Reply:
idk tbh 😄 I was just following the original pattern from v1 which does this as well. Afraid to remove it and cause a regression. WDYT should we try removing it?